api-registration/server/server.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import 'dotenv/config';
+import pg from 'pg';
+import argon2 from 'argon2';
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+app.use(express.json());
+
+app.post('/api/auth/sign-up', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password) {
+      throw new ClientError(400, 'username and password are required fields');
+    }
+    /* TODO:
+     * Hash the user's password with `argon2.hash()`
+     * Insert the user's "username" and "hashedPassword" into the "users" table.
+     * Respond to the client with a 201 status code and the new user's "userId", "username", and "createdAt" timestamp.
+     * Catch any errors.
+     *
+     * Hint: Insert statements can include a `returning` clause to retrieve the insterted row(s).
+     */
+    const hashedPassword = await argon2.hash(password);
+    const params = [username, hashedPassword];
+    const sql = `
+     insert into "users" ("username", "hashedPassword")
+     values($1, $2)
+     returning "userId", "username", "createdAt";
+     `;
+    const results = await db.query(sql, params);
+    const grade = results.rows;
```

Comment:
This isn't correct. It returns an array. You need to return just the single user.